### ISSUE TITLE
Error Creating Function with DynamoStream Event

### ISSUE THREAD

**Issue by beldougie** [localstack/localstack#2617]
# Type of request: This is a ...

[x] bug report
[ ] feature request

# Detailed description
Using serverless & the localstack/localstack:latest docker image (released 12 hours ago at time of creating this issue according to docker-hub), a previously correctly working cloudformation stack that creates a DynamoDB table with a DynamoStream that feeds the event source of a lambda function fails to deploy if we use the :latest version of the docker image.
...

## Expected behavior
It should deploy the stack to localstack correctly.
...

## Actual behavior
It does not deploy the stack correctly, and serverless raises an error 'Invalid EventSourceArn'. This goes away if I pin mydocker image to version 0.11.2 and the stack is deployed correctly.
...

# Steps to reproduce

We use docker compose to run localstack. Our docker-compose file looks like:

```
version: '2.1'

services:
  localstack:
    container_name: "${LOCALSTACK_DOCKER_NAME-localstack_main}"
    image: localstack/localstack
    ports:
      - "4566-4599:4566-4599"
      - "${PORT_WEB_UI-8080}:${PORT_WEB_UI-8080}"
    environment:
      - SERVICES=${SERVICES- }
      - DEBUG=${DEBUG- }
      - DATA_DIR=${DATA_DIR- }
      - PORT_WEB_UI=${PORT_WEB_UI- }
      - LAMBDA_EXECUTOR=${LAMBDA_EXECUTOR- }
      - KINESIS_ERROR_PROBABILITY=${KINESIS_ERROR_PROBABILITY- }
      - DOCKER_HOST=unix:///var/run/docker.sock
      - HOST_TMP_FOLDER=${TMPDIR}
      - LAMBDA_REMOTE_DOCKER=${LAMBDA_REMOTE_DOCKER-}
    volumes:
      - "${TMPDIR:-/tmp/localstack}:/tmp/localstack"
      - "/var/run/docker.sock:/var/run/docker.sock"
```

We use a script that calls `docker-compose up -d`

We then call a standard `serverless deploy` command to deploy our stacks into localstack to create the application. Everything worked perfectly until this morning when one of my colleagues decided to prune his docker images and, therefore pulled the latest image down. Of course, the fact that the failure was caused by localstack wasn't immediately obvious, but this has cost an entire man-day to track down the problem. 

Please revert the latest image to 0.11.2 until you have it working.

**Comment by whummer** [comment]
Thanks for reporting @beldougie . Can you please provide a small, isolated example to help us reproduce this issue? Ideally, if you could provide a minimal `serverless.yml` configuration that exposes this problem, that would be great. Thanks for your help.

**Comment by beldougie** [comment]
Hey @whummer no problem. Here you go:

The process I have used to recreate is:

```sh
# install serverless globally
npm i -g serverless

# create a new serverless skeleton
sls create -t aws-nodejs -p lsdemo -n demo

# configure node and install dependencies
cd lsdemo
npm init -y
npm i -D serverless-localstack

# !! Update the generated serverless yml with the example posted before

# start localstack using docker-compose example below (using latest localstack/localstack image
docker-compose up

# attempt deployment
sls deploy -v
```


## Serverless File used:
```yml
service: demo

provider:
  name: aws
  runtime: nodejs12.x
  stage: local
  region: us-east-1
  iamRoleStatements: 
    - Effect: Allow
      Action:
        - dynamodb:*
      Resource:
        Fn::GetAtt:
          - myDynamoTable
          - Arn
  environment:
    STAGE: local
    LAMBDA_REMOTE_DOCKER: 'false'
custom:
  localstack:
    stages:
      - local
    autostart: false
    lambda:
      mountCode: true
    docker:
      sudo: true
functions: 
  sendToGPHandler: 
    handler: src/handlers/test.handler
    events:
      - stream:
          type: dynamodb
          arn:
            Fn::GetAtt:
              - myDynamoTable
              - StreamArn
          batchSize: 1 # this should be increased to reduce load...
          MaximumBatchingWindowInSeconds: 5
          startingPosition: LATEST
          maximumRetryAttempts: 10
          enabled: true
resources: 
  Resources:
    myDynamoTable:
      Type: AWS::DynamoDB::Table
      Properties:
        TableName: testTable
        AttributeDefinitions:
          - AttributeName: Id
            AttributeType: S
          - AttributeName: Version
            AttributeType: N
        KeySchema:
          - AttributeName: Id
            KeyType: HASH
          - AttributeName: Version
            KeyType: RANGE
        ProvisionedThroughput:
          ReadCapacityUnits: 1
          WriteCapacityUnits: 1
        StreamSpecification:
          StreamViewType: NEW_IMAGE
              

plugins:
  - serverless-localstack
```

## Docker Compose file used
```yml
version: '2.1'

services:
  localstack:
    container_name: "${LOCALSTACK_DOCKER_NAME-localstack_main}"
    image: localstack/localstack
    ports:
      - "4566-4599:4566-4599"
      - "${PORT_WEB_UI-8080}:${PORT_WEB_UI-8080}"
    environment:
      - SERVICES=${SERVICES- }
      - DEBUG=${DEBUG- }
      - DATA_DIR=${DATA_DIR- }
      - PORT_WEB_UI=${PORT_WEB_UI- }
      - LAMBDA_EXECUTOR=${LAMBDA_EXECUTOR- }
      - KINESIS_ERROR_PROBABILITY=${KINESIS_ERROR_PROBABILITY- }
      - DOCKER_HOST=unix:///var/run/docker.sock
      - HOST_TMP_FOLDER=${TMPDIR}
      - LAMBDA_REMOTE_DOCKER=${LAMBDA_REMOTE_DOCKER-}
    volumes:
      - "${TMPDIR:-/tmp/localstack}:/tmp/localstack"
      - "/var/run/docker.sock:/var/run/docker.sock"
```

If I use _this_ docker file (note the single change to force version 0.11.2 of localstack), the deployment will succeed

```yml
```yml
version: '2.1'

services:
  localstack:
    container_name: "${LOCALSTACK_DOCKER_NAME-localstack_main}"
    image: localstack/localstack:0.11.2
    ports:
      - "4566-4599:4566-4599"
      - "${PORT_WEB_UI-8080}:${PORT_WEB_UI-8080}"
    environment:
      - SERVICES=${SERVICES- }
      - DEBUG=${DEBUG- }
      - DATA_DIR=${DATA_DIR- }
      - PORT_WEB_UI=${PORT_WEB_UI- }
      - LAMBDA_EXECUTOR=${LAMBDA_EXECUTOR- }
      - KINESIS_ERROR_PROBABILITY=${KINESIS_ERROR_PROBABILITY- }
      - DOCKER_HOST=unix:///var/run/docker.sock
      - HOST_TMP_FOLDER=${TMPDIR}
      - LAMBDA_REMOTE_DOCKER=${LAMBDA_REMOTE_DOCKER-}
    volumes:
      - "${TMPDIR:-/tmp/localstack}:/tmp/localstack"
      - "/var/run/docker.sock:/var/run/docker.sock"
```
```

**Comment by whummer** [comment]
Thanks for the detailed report @beldougie . We were able to reproduce this issue based on the example you've provided. This should be fixed in #2637 - can you please give it a try with the `latest` version of the Docker image? Please report here if the problem persists. Thanks

**Comment by beldougie** [comment]
@whummer nice work - the latest image works with our stack again. Thanks for the quick fix!
